tests/phpunit/tests/rest-api/rest-comments-controller.php

```diff
@@ -3597,4 +3616,395 @@ public static function data_head_request_with_specified_fields_returns_success_r
 			'get_items request' => array( '/wp/v2/comments' ),
 		);
 	}
+
+	/**
+	 * Create a test post with note.
+	 *
+	 * @param int $user_id Post author's user ID.
+	 * @return int Post ID.
+	 */
+	protected function create_test_post_with_note( $role ) {
+		$user_id = self::$user_ids[ $role ];
+		$post_id = self::factory()->post->create(
+			array(
+				'post_title'   => 'Test Post for Notes',
+				'post_content' => 'This is a test post to check note permissions.',
+				'post_status'  => 'contributor' === $role ? 'draft' : 'publish',
+				'post_author'  => $user_id,
+			)
+		);
+
+		for ( $i = 0; $i < self::$num_notes; $i++ ) {
+			self::factory()->comment->create(
+				array(
+					'comment_post_ID'  => $post_id,
+					'comment_type'     => 'note',
+					'comment_approved' => 0 === $i % 2 ? 1 : 0,
+				)
+			);
+		}
+
+		return $post_id;
+	}
+
+	public function test_cannot_read_note_without_post_type_support() {
```

Comment:
Let's add `@ticket 64096` comments to the new test cases and related methods. See examples: https://github.com/WordPress/wordpress-develop/blob/4ab0abf3767b597919ee52582eb4e0dcbdcda82f/tests/phpunit/tests/rest-api/rest-comments-controller.php#L246-L248